[Chore] style 초기화

HyeonChae1104 opened the pull request:
## 주요 변경사항
- reset-css를 설정하여 기존의 스타일을 초기화하였습니다. 
- breakpoints를 정의하고 적용하였습니다. (styles/variants/index.ts)

## 리뷰어에게...
부족한 부분이 있다면 피드백 부탁드립니다! 

## 관련 이슈
closes #4

HyeonChae1104 committed:
chore: reset-css 설정

기존의 style을 초기화합니다.

#3

HyeonChae1104 committed:
chore: breakpoints 정의 및 적용

breakpoints를 정의하여 다양한 반응형 레이아웃을 지원합니다.

#3

HyeonChae1104 committed:
chore(styles): reset.css 적용 및 variants export 설정

reset.css를 통해 브라우저 기본 스타일을 초기화하고,
variants 폴더의 변수를 vars로 export하여 다른 파일에서 사용할 수 있도록 설정합니다.

#3

geongyu09 reviewed (changes requested):
올바르게 reset-css가 적용되지 않은 것 같습니다..! 
reset.css 파일은 올바르게 만들었지만, 해당 파일이 적용이 안되고 있습니다

![image](https://github.com/user-attachments/assets/3a5a0c63-aeb5-4913-8a7c-0f716e71d570)
h1 태그를 추가해보았는데요. reset-css가 올바르게 적용되었다면 위에 있는 일반 a, p 태그와 크기 차이가 있으면 안됩니다!

index.css 부분과 app.css 부분을 올바르게 수정하고, 실제 ts/tsx 파일에 적용되도록 수정해야 할 것 같습니다!

HyeonChae1104 committed:
chore: reset-css 적용

reset-css를 import하여 적용하였습니다.

#3

geongyu09 commented on src/App.css:
```diff
@@ -1,3 +1,4 @@
+@import "./styles/reset.css";
```
단순히 저희가 reset.css 파일을 import 하면 지금것 만들었던 계층 구조가 깨지게 됩니다.
현재 저희는 만들어질 수 있는 css 파일을 전부 styles에 넣어두고, 이를 /styles/index.ts에 import 하고 있습니다. 
즉 모든 css에 대한 정보는 index.ts를 거쳐서 나가도록 하고 있어요. 
그리고 index.ts만 실제 저희 프로젝트에 적용하고 있죠.
덕분에 관리가 정말 편하고 저희 프로젝트에서는 css에 대해서 큰 관심을 가지지 않고 react 로직에만 집중할 수 있게 해줍니다. 

해당 로직은 `css-> index.ts ->프로젝트`  단계대로 작성된 것이 아니라 `reset.css -> App.css -> 프로젝트` 로 가게 되는 로직인 것 같습니다. 
수정이 필요해보입니다. 

아래 절차대로 하면 됩니다. 

1. App.css, index.css 제거
2. /src/index.tsx에 `import '@/styles';` import 문 추가
  (헌재 craco 이슈가 있으므로 `import './styles/index';` 로 대체해도 좋습니다.)

Da-Yeon-Kim replied:
오, 제가 하고 싶은 말을 건규 씨가 이미 엄청 친절하게 작성해주셨네요! ㅎㅎ
건규 씨가 말씀해주신 부분들은 2단계 프로젝트와 유사해서
그때 코드를 좀 참고하시면, 폴더 구조나 로직에 대한 이해가 더 수월하실 거예요~

HyeonChae1104 replied:
꼼꼼하게 남겨주셔서 감사합니다! 잘 읽어보고 알려주신대로 수정해보았습니다.!

geongyu09 commented on src/styles/reset.css:
```diff
@@ -0,0 +1,99 @@
+html {
```
추가적으로 확인해보니 해당 reset.css 로직에도 font-size 초기화 로직이 빠지는 등 문제가 있는 것 같습니다. 
아래의 로직으로 변경 부탁드립니다! 

<details>
<summary>
reset.css 로직
</summary>
html, body, div, span, applet, object, iframe,
h1, h2, h3, h4, h5, h6, p, blockquote, pre,
a, abbr, acronym, address, big, cite, code,
del, dfn, em, img, ins, kbd, q, s, samp,
small, strike, strong, sub, sup, tt, var,
b, u, i, center,
dl, dt, dd, ol, ul, li,
fieldset, form, label, legend,
table, caption, tbody, tfoot, thead, tr, th, td,
article, aside, canvas, details, embed, 
figure, figcaption, footer, header, hgroup, 
menu, nav, output, ruby, section, summary,
time, mark, audio, video {
	margin: 0;
	padding: 0;
	border: 0;
	font-size: 100%;
	font: inherit;
	vertical-align: baseline;
}
/* HTML5 display-role reset for older browsers */
article, aside, details, figcaption, figure, 
footer, header, hgroup, menu, nav, section {
	display: block;
}
body {
	line-height: 1;
}
ol, ul {
	list-style: none;
}
blockquote, q {
	quotes: none;
}
blockquote:before, blockquote:after,
q:before, q:after {
	content: '';
	content: none;
}
table {
	border-collapse: collapse;
	border-spacing: 0;
}
</details>

HyeonChae1104 replied:
해당 로직으로 변경하였습니다!

geongyu09 reviewed (commented):
몇가지 더 남겨보았습니다! 
저희가 왜 이런 폴더구조를 가지고, 왜 index.ts 파일이 존재하는지에 대해서 조금 더 생각해봐도 좋을 것 같습니다! 

변경이 어려우시면 언제든 연락해주세요! 
(질문 사항이 있으면 해당 pr에 마구마구 남겨주세요. 문제를 함께 소통하며 해결하려고 깃허브가 존재하니까요!)
그리고 pr에 변경사항이 생긴다면 오른쪽 위에 리뷰어 설정 부분에 화살표로 원 그려진 부분 클릭하면 re-request 할 수 있어요. 변경 사항 생기면 꼭 눌러주세요! 
(안누르면 메일 알림도 안오고, 저도 커맨드밖에 달지 못해요!)

HyeonChae1104 committed:
chore: App.css, index.css 제거 및 스타일 관리 구조 개선

- App.css, index.css 파일 삭제
- src/index.tsx에 import '@/styles' 추가
- 현재 craco 이슈로 인해 import './styles/index'로 대체

#4

HyeonChae1104 committed:
chore: reset.css 로직 변경

몇가지 초기화 로직이 누락되어 해당 로직으로 변경하였습니다.

#4

HyeonChae1104 committed:
chore: 절대 경로 설정

@styles 사용을 위한 절대 경로를 설정하였습니다.

#4

geongyu09 commented on src/index.tsx:
```diff
@@ -1,6 +1,8 @@
+import '@styles/index'
```
`@/styles/index` 로 수정해야합니다

HyeonChae1104 replied:
헉.. 넵..!

Da-Yeon-Kim replied:
그리고 index는 자동으로 인식되니까 굳이 안 적으시고 `@/styles`만 적으셔도 되어요:D
명시하셔도 큰 상관은 없지만, 보통 적지 않으니 통일성을 주는 게 좋을 것 같긴 해요~

geongyu09 reviewed (changes requested):
한번 더 수정해야할 것 같아요..

HyeonChae1104 committed:
chore: import문 수정

1. craco 설정에서 절대 경로를 지원하지 않아서 발생하는 문제 해결 후 import 문 수정하였습니다.

#4

geongyu09 commented on src/index.tsx:
```diff
@@ -17,3 +19,5 @@ root.render(
 // or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
 reportWebVitals();
+
+export {};
```
해당 로직은 불필요해보입니다! 제거해도 좋아보입니다

geongyu09 commented on src/index.tsx:
```diff
@@ -1,6 +1,8 @@
+import "@/styles"
```
👍👍👍👍

geongyu09 reviewed (approved):
reset-css가 올바르게 적용이 되었네요! 
추가적으로 수정할 부분이 있긴 있어보이지만, 중요한 사항은 아니므로 approve 하였습니다!

craco, react-router-dom 의존성 추가는 이미 merge가 되었던 다른 pr에서 작업한 내용이라서 꼭 필요한 작업은 아니었다고 생각합니다! 
이러한 것들은 오히려 컨플릭이나 기존에 해결되었던 문제들을 다시 일으킬 수도 있고 버전이 다를 수도 있게 되므로 주의가 필요합니다. 
해결하려는 이슈와 다른 성격을 가지고 있기도 하므로 이러한 것들을 꼭 해당 pr 내용에 담아야 했다면, 그 이유를 다른 피어분들에게 충분히 설명해줄 필요도 있어보입니다! 

고생많으셨습니다! 그대로 머지해도 괜찮아 보이지만 시간적 여유가 된다면 src/index.tsx 부분 수정해주세요~~!

geongyu09 commented:
컨플릭 해결하게 되면 커밋이 하나 쌓이게 됩니다! 
해당 커밋도 가능하다면 컨밴션을 지켜주세요!

HyeonChae1104 committed:
chore: export{} 삭제

불필요한 로직을 삭제하였습니다.

#4

HyeonChae1104 committed:
Merge branch 'Weekly' into chore/3-reset-css
